tests/ui/lint/non-local-defs/cargo-update.stderr

```diff
@@ -1,14 +1,22 @@
-warning: non-local `impl` definition, they should be avoided as they go against expectation
+warning: non-local `impl` definition, `impl` blocks should be written at the same level as their item
   --> $DIR/cargo-update.rs:17:1
    |
 LL | non_local_macro::non_local_impl!(LocalStruct);
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
+   | |
+   | `LocalStruct` is not local
+   | `Debug` is not local
```

Comment:
We should customize the error when involving macros, because the suggestion to move the impl out of a const is not actionable for users in this case. Even further, it is likely that we shouldn't emit the lint if it comes from a foreign macro (do not deal with that in this PR).

Reply:
I agree it would be good if we could have a specialized version of the diagnostic for macro-generated code, but I have no idea what it would look like.

> Even further, it is likely that we shouldn't emit the lint if it comes from a foreign macro (do not deal with that in this PR).

We must still emit the warning in macro-generated code, since the goal is to make the lint deny-by-default and then hard-error in a future edition, and without the warn it would go straight to hard-error which is not good.

Reply:
As follow up work (not this PR), we should still customize the error to state "the macro needs to be changed".